### PR TITLE
[MIG] privacy_consent: migrate to 16.0

### PR DESCRIPTION
**Pull request by tde-banana-odoo**

### CODE REVIEW

**Comment by tde-banana-odoo** [comment]
Waiting https://github.com/OCA/data-protection/pull/70

**Comment by pedrobaeza** [comment]
/ocabot migration privacy_consent

You can now rebase the PR after merging the dependent one.

**Comment by tde-banana-odoo** [comment]
@pedrobaeza CIs are yelling about development_status that is incompatible (beta for privacy, stable for privacy_content). I updated privacy to be in "Production/Stable", is it problematic ?

**Comment by pedrobaeza** [comment]
Oh, indeed the inconsistency came from where the linter was not activated. Can you please isolate the development status promotion into a separate commit `[IMP] privacy: Promote to stable` or similar?

**Comment by tde-banana-odoo** [comment]
Hello,

There was a "hackish" fix to do due to removal of t-raw when displaying the html version of a partner's address . Issue is that we have to markup the html content before rendering it, we cannot directly inject html in templates anymore. I made a specific commit as it is not obvious, and maybe deserves its own review https://github.com/OCA/data-protection/pull/71/commits/c90d4dd6df0c77c7b850355a3d904dd54e5aa3ad .

File: privacy_consent/views/res_partner.xml [hunk 1]
```diff
@@ -6,16 +6,13 @@
...
-            name="groups_id"
-            eval="[(4, ref('privacy.group_data_protection_user'))]"
-        />
         <field name="arch" type="xml">
             <div name="button_box" position="inside">
                 <button
                     attrs='{"invisible": [("privacy_consent_count", "=", 0)]}'
                     class="oe_stat_button"
                     context='{"search_default_partner_id": active_id}'
+                    groups="privacy.group_data_protection_user"
```
Comment (yajo) [inline privacy_consent/views/res_partner.xml:15]: The good part of adding the permission in the view itself is that it won't overload the requests with new useless fields. And some times it is even necessary to avoid bugs, such as when the field requires extra permissions that the user may not have. For instance, to compute this count, you need read access to consents if I'm not wrong.

So, I think you should revert this change.

**Review by yajo — CHANGES_REQUESTED** [review]
Pretty good migration in general. Just one minor thing missing.

File: privacy_consent/views/res_partner.xml (hunk 1 above)
Comment (tde-banana-odoo) [reply privacy_consent/views/res_partner.xml:15]: Groups cannot be applied on inherited views since odoo v16 (only on base views), allowing to simplify inheritance computation (all inherited views always apply, maybe with a void diff but always present). That's why I moved it on the field definition itself.

Comment (yajo) [reply privacy_consent/views/res_partner.xml:15]: But then, how can you overcome the permission problems?

Comment (tde-banana-odoo) [reply privacy_consent/views/res_partner.xml:15]: With groups on the field it will not try to compute it. The inherit is present in the hierarchy but not the field itself. Globally idea is to move groups on view inherits to part of the arch that we want to protect (even if it consists in the whole arch of the inherit).

Comment (tde-banana-odoo) [reply privacy_consent/views/res_partner.xml:15]: For more details, you can check work of Denis at https://github.com/odoo/odoo/pull/98551 and https://github.com/odoo/odoo/pull/95729 notably :)

Comment (pedrobaeza) [reply privacy_consent/views/res_partner.xml:15]: @yajo can you please re-check this?

**Review by pedrobaeza — APPROVED** [review]

**Review by yajo — APPROVED** [review]
